### ISSUE TITLE
Refactor code

### ISSUE THREAD

**Issue by davidinouye** [davidinouye/automated-dependence-plots#1]
First stage:
- [x] Tag master as v0.0.1
- [x] Merge in changes from master into refactor
- [x] * Remove references to deprecated "counterfactual" module
- [x] Remove "counterfactual" link
- [x] Add .gitignore file
- [x] Add cached model to zips of cached models
- [x] Clean up examples to use helper functions (e.g., bias figure)
- [x] * Update code to work on most recent scikit-learn and PDP instead of monkey patch 
- [x] * Removed monkey patch and just use sklearn's pdp function... also cleaned up figure-loan-model-comparions
- [x] * Remove deprecated utils.py
- [x] * Add Lipschitz utility based on notebook
- [x] Remove deprecated "GlobalModelUtility" maybe edit to "ModelContrast" instead of "ModelComparison"
- [x] Replace constant things with Least/MostConstantUtility
- [x] Remove old aliases in Utility functions including "OldMonotonicUtility"
- [x] Update models zip and url in Makefile. 
- [x] Need to check errors of streetsign figure and maybe others (checked other notebooks in results though)
- [x] Changed name from BestPartial to FunctionalPropertyUtility
- [x] Remove auxiliary_scoring and use metric instead of scoring
- [x] Check for scoring everywhere
- [x] Double check utilities
- [x] * Add functionality to call utility using y values and optionally x vals instead of curve
- [x] Add to README: main functionality : Curves (directional, categorical, and bounds), Utilities (constant auto-curve-target), Optimization, and Plotting.
- [x] Rerun test suite to make sure it all still works
- [x] Check that new result notebooks are correct (i.e., have reasonable figures)
- [x] Copy over new notebooks from results
- [x] Create new release
- [x] Email group about changes

Update to sklearn 0.23.1 (newest stable version):
- [x] scorer inside utility: FutureWarning: The sklearn.metrics.scorer module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.metrics. Anything that cannot be imported from sklearn.metrics is now part of the private API.
- [x] (in helpers.py fo train_test_split? or CV?) /Users/dinouye/anaconda3/envs/adp-env-new-sklearn/lib/python3.7/site-packages/sklearn/model_selection/_split.py:297: FutureWarning: Setting a random_state has no effect since shuffle is False. This will raise an error in 0.24. You should leave random_state to its default (None), or set shuffle=True.

**Comment by davidinouye** [comment]
Done and done with v0.0.2. :-)
